gp-includes/strings.php

```diff
@@ -96,21 +96,23 @@ function gp_levenshtein( $str1, $str2, $length1, $length2 ) {
 		return 0;
 	}
 
-	$bytelength1 = strlen( $str1 );
-	$bytelength2 = strlen( $str2 );
+	if ( $length1 <= 255 && $length2 <= 255 ) {
+		$bytelength1 = strlen( $str1 );
+		$bytelength2 = strlen( $str2 );
 
-	if ( $bytelength1 === $length1 && $bytelength1 <= 255
-		 && $bytelength2 === $length2 && $bytelength2 <= 255 ) {
-		return levenshtein( $str1, $str2 );
+		if ( $bytelength1 === $length1 && $bytelength2 === $length2 ) {
+			return levenshtein( $str1, $str2 );
+		}
 	}
 
+	$chars1 = preg_split( '//u', $str1, -1, PREG_SPLIT_NO_EMPTY );
+	$chars2 = preg_split( '//u', $str2, -1, PREG_SPLIT_NO_EMPTY );
```

Comment:
Per @ocean90 above:
```suggestion
	$chars1 = mb_str_split( $str1 );
	$chars2 = mb_str_split( $str2 );
```

PHP docs suggest passing encoding, but I'm not sure if we'd want to hard-code utf8?
```suggestion
	$chars1 = mb_str_split( $str1, 1, 'UTF-8' );
	$chars2 = mb_str_split( $str2, 1, 'UTF-8' );
```

Reply:
> PHP docs suggest passing encoding, but I'm not sure if we'd want to hard-code utf8?

This seems to be true for all mb_* functions. Since neither GlotPress nor WordPress are passing encoding to other functions I think we can continue to omit it.

I'll handle the PHP bump in a separate PR shortly.